frontend/src/main/scala/bloop/exec/JvmProcessForker.scala

```diff
@@ -112,22 +112,111 @@ final class JvmForker(config: JdkConfig, classpath: Array[AbsolutePath]) extends
-      val logTask = if (logger.isVerbose) {
-        val debugOptions =
-          s"""
-             |Fork options:
-             |   command      = '${cmd.mkString(" ")}'
-             |   cwd          = '$cwd'""".stripMargin
-        Task(logger.debug(debugOptions)(DebugFilter.All))
-      } else Task.unit
-      logTask.flatMap(_ => Forker.run(cwd, cmd, logger, opts))
+      val cmdStr = cmd.mkString(" ")
+
+      // Note that we current only shorten the classpath portion and not other options
+      // Thus we do not yet *guarantee* that the command will not exceed OS limits
+      if (cmdStr.length > processCmdCharLimit) {
+        def charLimitMsg =
+          s"""|Supplied command to fork exceeds character limit of $processCmdCharLimit
+              |Creating a temporary MANIFEST jar for classpath entries
+              |""".stripMargin
+
+        logDebugIfVerbose(charLimitMsg).flatMap { _ =>
+          withTempManifestJar(fullClasspath, logger) { manifestJar =>
+            val shortClasspathOption = "-cp" :: manifestJar.syntax :: Nil
+            val shortCmd = java.syntax :: jvmOptions.toList ::: shortClasspathOption ::: appOptions
+            runCmd(shortCmd)
+          }
+        }
+      } else {
+        runCmd(cmd)
+      }
+    }
+  }
+
+  private def withTempManifestJar[A](
+      classpath: Array[AbsolutePath],
+      logger: Logger
+  )(op: AbsolutePath => Task[A]): Task[A] = {
+
+    val manifestJar = Files.createTempFile("jvm-forker-manifest", ".jar").toAbsolutePath
+    val manifestJarAbs = AbsolutePath(manifestJar)
+    def cleanup = {
+      Task {
+        if (logger.isVerbose) {
+          logger.debug(s"Cleaning up temporary MANIFEST jar: $manifestJar")(DebugFilter.All)
+        }
+        Paths.delete(manifestJarAbs)
+      }
+    }
+
+    val classpathStr = classpath.map(addTrailingSlashToDirectories).mkString(" ")
+
+    val manifest = new Manifest()
+    manifest.getMainAttributes.put(Attributes.Name.MANIFEST_VERSION, "1.0")
+    manifest.getMainAttributes.put(Attributes.Name.CLASS_PATH, classpathStr)
+
+    val out = Files.newOutputStream(manifestJar)
+    // This needs to be declared since jos itself should be set to close as well.
+    var jos: JarOutputStream = null
+    try {
+      jos = new JarOutputStream(out, manifest)
+    } finally {
+      if (jos == null) {
+        out.close()
+      } else {
+        jos.close()
+      }
+    }
+
+    op(manifestJarAbs)
+      .doOnFinish(_ => cleanup)
+      .doOnCancel(cleanup)
+  }
+
+  // Turns out manifest files can use absolute path directories in the classpath
+  // But they need a trailing slash
+  private def addTrailingSlashToDirectories(path: AbsolutePath): String = {
+    val syntax = path.syntax
+    val separatorAdded = if (syntax.endsWith(".jar")) {
+      syntax
+    } else {
+      syntax + File.separator
     }
+    // On Windows, the drive letter is prepended to absolute paths
```

Comment:
Thank you for tracking down the windows failures and fixing the issue! It’s great to see the CI green 😁 

Should we maybe guard this logic for Windows only? Im not sure if Unix file systems support file names with colons.